### PR TITLE
Tests, Vignettes and R CMD Check

### PR DESCRIPTION
**Pull request by Yuri05**
This PR introduces no **functional** changes. All changes to the **/R**-files are due to adding missing documentation or fixing R CMD Check warnings, such as fully qualifying called functions.

Changes/Additions made by the PR:

## 1. Added the package vignettes based on the accompanying publication ([Najjar et al., 2024](https://doi.org/10.1002/psp4.13256)) (mainly using the information and examples from its supplement)

<details><summary>Details</summary>

The package shipped with placeholder pkgdown article config and no vignettes. This adds four vignettes derived from the supplementary materials of the accompanying publication ([Najjar et al., 2024](https://doi.org/10.1002/psp4.13256)) and integrates them into the site.

Content was sourced from the article's supplementary `.docx` (retrieved via Europe PMC after the Wiley download was Cloudflare-blocked) and converted with `pandoc`, preserving equations as LaTeX and extracting embedded figures.

### Vignettes (`vignettes/`)
- **`getting-started.Rmd`** — Suppl. Materials 3: install, workflow overview, parameter/output specification, OAT + Morris/Sobol/EFAST runs, Shiny GUI. Includes workflow + GUI figures.
- **`uv-filter-formulation.Rmd`** — Suppl. Materials 5: octocrylene/oxybenzone walkthrough, parameter table (S4), result interpretation.
- **`midazolam-ddi.Rmd`** — Suppl. Materials 6: rifampicin–midazolam DDI-ratio analysis, parameter table (S5), interpretation.
- **`variance-based-methods.Rmd`** — Suppl. Materials 2: full derivation with MathJax-rendered equations, Figure S1 (sampling) and S2 (EFAST) figures.

All example code chunks are `eval=FALSE`, mirroring the supplement, so the build does not require the OSP simulation toolchain or PKML assets.

### Integration / supporting
- `_pkgdown.yml`: replaced commented stub with an `articles:` section grouping vignettes into *Get started*, *Examples*, *Background*.
- `DESCRIPTION`: added `VignetteBuilder: knitr` (`knitr`/`rmarkdown` already in Suggests).
- `inst/WORDLIST`: added domain terms, author names, and LaTeX tokens so the existing spell check passes.
- `vignettes/figures/`: 12 PNGs extracted from the supplement.

### Notes for reviewers
- The Figure S1 composite (8-cell table in the source) is reconstructed as a 3×2 quantile→parameter-space figure grid.
- WORDLIST now contains LaTeX fragments (`cdot`, `bmatrix`, …) flagged inside the math vignette; an alternative is disabling vignette spell-checking instead.

</details> 

## 2. Added more unit tests based on the just created vignettes and the article

<details><summary>Details</summary>

### Summary
Added 3 new test files and extended 6 existing ones, covering previously untested functions:

#### New test files

- `test-lowry-functions.R` — Lowry plots and Sobol/EFAST bar graphs, using the n = 3 parameter example from the variance-based-methods vignette (first-order indices summing to < 1 so interactions are visible, `S_i^T ≥ S_i`). Covers the single-parameter edge case and the descending first-order ordering.
- `test-tornado-functions.R` — one-at-a-time local sensitivity/uncertainty tornado plots as used by `runSU()` in the Getting Started vignette: nested plot structure, absolute-magnitude ordering, sign colouring, `NA → 0` coercion, and the uncertainty-mode warning/`NULL` branch.
- `test-contour-plot.R` — the EFAST response-surface helpers' data-sufficiency/variance-check branches (which don't need the optional `interp `back-end).

#### Extended files

- `test-morris-functions.R` — `generateMorrisPlot:` μ*/σ axes, rankingNorm ordering, log10 transform (verified via `ggplot_build`).
- `test-efast-functions.R` — quantile→parameter-space mapping (Figure 1 inverse-CDF), the documented `½ + (1/π)·arcsin(sin(…))` EFAST search-curve (Saltelli et al. 1999, phase reproduced via seed), and multi-output `getEFASTResultsDf`.
- `test-sampling-functions.R` — the quantile-transformation wrappers, `*Inputs` helpers, and `quantileTransformationFunctions `registry, grounded in vignette distributions (LogUniform[2,8], LogNormal mean=10/CV=0.5).
- `test-su-functions.R` — `cleanUpSUDf` grouping/uncertainty-column placement.
- `test-utilities.R` — `buildSettingsCMD `settings reporting (distribution, parameters, units, PK selections).
- `test-distribution.R` — vignette LogNormal median, monotonicity across the unit interval, LogUniform non-numeric rejection.

</details> 

## 4. Fix R CMD Check warnings (missing documentation, spelling, etc.)

<details><summary>Details</summary>

`R CMD check` reported numerous "no visible global function definition" and "no visible binding for global variable" NOTEs because functions were used without being imported into the `NAMESPACE`, and bare column names were referenced inside non-standard-evaluation calls (`aes()`, `subset()`, `facet_grid()`).

### Changes

- **`R/ospsuite.globalsensitivity-package.R`** (new): package-level doc holding all import declarations and a `utils::globalVariables()` registration.
  - `@importFrom` for `stats` (`aggregate`, `approx`, `fft`, `na.omit`, `qnorm`, `quantile`, `runif`, `sd`), `utils` (`head`, `tail`), `dplyr` (`%>%`), `ospsuite` (`pkAnalysesToDataFrame`); `@import shiny` to cover the GUI functions in `app.R`.
  - `globalVariables()` for NSE column names (`mustar`, `stdv`, `label`, `ParameterDisplayName`, `values`, `signColumn`, `x`/`y`/`z`, `xLab`/`yLab`, `output`, `pk`, `Parameter`, `Value`, `Measure`, `ParameterRanks`, `ymin`/`ymax`/`ymean`).
- **`R/output.R`**: qualify the exported enum as `ospsuite::StandardPKParameter`.
- **`NAMESPACE`**: add the corresponding `import`/`importFrom` directives (matching roxygen2 output ordering).
- **`DESCRIPTION`**: add `utils` to `Depends` (now imported from).
- **`man/ospsuite.globalsensitivity-package.Rd`** (new): generated package-level man page for the `"_PACKAGE"` block.

Coverage extends slightly past the (truncated) issue list — `sd`/`quantile`/`na.omit` and `pk`/`xLab`/`yLab` — so the remaining NOTEs in the same functions are also cleared.

```r
## usethis namespace: start
#' @import shiny
#' @importFrom dplyr %>%
#' @importFrom ospsuite pkAnalysesToDataFrame
#' @importFrom stats aggregate approx fft na.omit qnorm quantile runif sd
#' @importFrom utils head tail
## usethis namespace: end
NULL

utils::globalVariables(c("Measure", "Parameter", "mustar", "stdv", "label", ...))
```

Note: `NAMESPACE` was edited by hand to mirror roxygen2's output since R/roxygen2 isn't available in this environment; re-running `devtools::document()` should reproduce it.

R CMD check reported a NOTE for undefined globals in `runGUI`'s server (`getBaseUnit`, `ospDimensions`, `toUnit`, `ospUnits`, `getSimulationTree`, `renderTree`, `shinyTree`, `zip`) plus documentation WARNINGs for the undocumented `distribution` object and an argument mismatch in `checkParametersExistInSimulation`.

### Changes

- **`R/app.R` — namespace-qualify cross-package symbols** in line with the file's existing style (`ospsuite::getParameter`, etc.):
  - `ospsuite::` → `getBaseUnit`, `ospDimensions`, `toUnit`, `ospUnits`, `getSimulationTree`
  - `shinyTree::` → `renderTree`, `shinyTree`
  - `utils::zip` (4 call sites)

  ```r
  # before
  ospParameterUnit <- getBaseUnit(ospParameterDimension)
  # after
  ospParameterUnit <- ospsuite::getBaseUnit(ospParameterDimension)
  ```

  Qualifying `zip` directly avoids the suggested `importFrom("utils","zip")`, matching the rest of the file.

- **`R/sampling-functions.R` / `man/distribution.Rd`** — add roxygen docs for the exported `distribution` list and generate the corresponding data `.Rd`.

- **`R/utilities.R` / `man/checkParametersExistInSimulation.Rd`** — correct the roxygen `@param` from the non-existent `outputPaths` to the actual `parameterPaths`, clearing both the undocumented-argument and documented-but-not-in-usage warnings.

### Scope note

Qualification is limited to `R/app.R` (the source R CMD check evaluates). Unqualified calls in `examples/` and `vignettes/` were not flagged and are left untouched.
</details> 

## 5. Add function reference to pkgdown

<details><summary>Details</summary>

The pkgdown site had no `reference` section, leaving only a commented-out placeholder, so exported functions were not organized on the package website.

### Changes
- **Added a grouped `reference` section** to `_pkgdown.yml` covering all 17 exported functions, replacing the placeholder comment.
- **Organized functions into logical sections** with titles and descriptions:
  - *Running sensitivity analyses*: `runSobol`, `runEFAST`, `runMorris`, `runSU`
  - *Inputs*: `SAParameter`, `SAOutput`
  - *Distributions*: `distribution`, `UniformDistribution`, `LogUniformDistribution`, `NormalDistribution`, `LogNormalDistribution`
  - *Visualization*: `generateSobolBarGraph`, `generateEFASTBarGraph`, `generateMorrisPlot`, `generateTornadoPlot`, `getContourPlot`
  - *Graphical user interface*: `runGUI`
</details> 

## Summary by CodeRabbit

* **New Features**
  * Added new user guides and examples, including getting started, case-study walkthroughs, and a methods article.
  * Expanded the website navigation to make articles easier to find.

* **Documentation**
  * Improved package and function help pages for sampling, simulation checks, and distribution references.

* **Bug Fixes**
  * Updated package setup so dependencies and utilities load more reliably.
  * Improved plot and analysis handling for edge cases, including sparse data and missing values.

* **Tests**
  * Added broader automated test coverage for sampling, sensitivity analyses, plot generation, and utility output.

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -22,12 +22,15 @@ Depends:
...
-    interp
+    interp,
+    utils
 Suggests: 
     knitr,
     rmarkdown,
     testthat (>= 3.0.0)
+VignetteBuilder: knitr
 Config/testthat/edition: 3
+Config/testthat/parallel: true
```
Comment (Yuri05) [inline DESCRIPTION:34]: At the moment all tests seem to run in parallel; in case we face any problems in the future - quick and dirty fix would be just setting this to **false**

File: inst/WORDLIST [hunk 2]
```diff
@@ -1,34 +1,91 @@
...
+timeprofiles
+TLF
+unicode
+varphi
+vdots
+vivo
+ymax
+ymin
+µm
+µmol
```
[the remark this answers is not in the document]
Comment (Yuri05) [reply inst/WORDLIST:92]: fixed

**Comment by Yuri05** [comment]
@abdullahhamadeh @rengelke Ready for review (The Windows job has failed for the same reason it has frequently failed in recent weeks, and this has nothing to do with the changes in this PR.)

**Review by abdullahhamadeh — APPROVED** [review]
